src/PsrLogAdapter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB;
+
+use MongoDB\Driver\Logging\Logger as DriverLogger;
+use Psr\Log\LoggerInterface as PsrLogger;
+use Psr\Log\LogLevel as PsrLogLevel;
+use SplObjectStorage;
+
+use function MongoDB\Driver\Logging\addLogger;
+use function MongoDB\Driver\Logging\removeLogger;
+
+final class PsrLogAdapter implements DriverLogger
+{
+    private static ?self $instance = null;
+
+    /** @psalm-var SplObjectStorage<PsrLogger, null> */
+    private SplObjectStorage $psrLoggers;
+
+    private const PSR_LEVELS = [
+        DriverLogger::LEVEL_ERROR => PsrLogLevel::ERROR,
+        /* libmongoc considers "critical" less severe than "error" so map it to
+         * "error" in the PSR logger. */
+        DriverLogger::LEVEL_CRITICAL => PsrLogLevel::ERROR,
+        DriverLogger::LEVEL_WARNING => PsrLogLevel::WARNING,
+        DriverLogger::LEVEL_MESSAGE => PsrLogLevel::NOTICE,
+        DriverLogger::LEVEL_INFO => PsrLogLevel::INFO,
+        DriverLogger::LEVEL_DEBUG => PsrLogLevel::DEBUG,
+        /* PSR does not define a "trace" level, so map it to "debug" in the PSR
+         * logger. That said, trace logging is very verbose and should not be
+         * encouraged when using a PSR logger. */
+        DriverLogger::LEVEL_TRACE => PsrLogLevel::DEBUG,
+    ];
+
+    public static function addLogger(PsrLogger $psrLogger): void
+    {
+        $instance = self::getInstance();
+
+        $instance->psrLoggers->attach($psrLogger);
+
+        addLogger($instance);
```

Comment:
You may skip this call if the logger is already registered.
```
        if ($instance->psrLoggers->count() === 0) {
            addLogger($instance);
        }
```

Reply:
PHPC handles this already. Since this is not in the hot path I think we can do without duplicating such logic here.